src/prg_quantumdynamics_mod.F90

```diff
@@ -52,104 +52,71 @@ subroutine prg_kick_density(kick_direc,kick_mag,dens,N,S,SINV,&
     type(bml_matrix_t)                          :: T1, T2, rho_bml, s_bml, sinv_bml
     character(len=*), intent(in)                :: bmltype
 
-    allocate (tmat1(N))
-    allocate (tmat2(N))
+    allocate (tmat1(norbs))
+    allocate (tmat2(norbs))
 
-    do i=1,N
+    do i=1,norbs
        telem = cmplx(0.0_dp,-kick_mag*r(which_atom(i),kick_direc))
        tmat1(i) = exp(telem)
        tmat2(i) = exp(-telem)
     enddo
 
-    call bml_zero_matrix(bmltype, BML_ELEMENT_COMPLEX,dp,N,N, T1)
+    call bml_zero_matrix(bmltype, BML_ELEMENT_COMPLEX,dp,norbs,M, T1)
     call bml_set_diagonal(T1,tmat1,thresh)
     deallocate (tmat1)
-    call bml_zero_matrix(bmltype, BML_ELEMENT_COMPLEX,dp,N,N, rho_bml)
-    call bml_convert_from_dense(bmltype,dens,rho_bml,thresh,N)
-    call bml_zero_matrix(bmltype, BML_ELEMENT_COMPLEX,dp,N,N, T2)
+    call bml_zero_matrix(bmltype, BML_ELEMENT_COMPLEX,dp,norbs,M, rho_bml)
```

Comment:
You don't have to allocate a new matrix before calling the import `bml_convert_from_dense`. The import function will deallocate the matrix and allocate a new one with the appropriate size.